protocols/kad/src/handler.rs

```diff
@@ -1051,6 +1054,41 @@ fn process_kad_response(event: KadResponseMsg, query_id: QueryId) -> HandlerEven
     }
 }
 
+struct WakeableVecDeque<T> {
+    inner: VecDeque<T>,
+    empty_waker: Option<Waker>,
+}
+
+impl<T> Default for WakeableVecDeque<T> {
+    fn default() -> Self {
+        Self {
+            inner: Default::default(),
+            empty_waker: Default::default(),
+        }
+    }
+}
+
+impl<T> WakeableVecDeque<T> {
+    fn push(&mut self, value: T) {
```

Comment:
Let's stick to the naming of the `VecDeque`.

```suggestion
    fn push_back(&mut self, value: T) {
```

Reply:
done.